### PR TITLE
Add milliseconds to dateutil.Time

### PR DESCRIPTION
**Pull request by paulbalomiri**
Hi,
I propose this PR for solving #90 

* Added sub second precision based on `dtstart%1000`
* All recurrences are offset from their respective interval start by `(dtstart - (dtstart%1000))`
* `dateutil.Time` has now a `getSubSeconds()`, and accepts an additional `sub_seconds` init argument
* Added testcases which test both :
  * the `plb(bysetpos) && plb(timeset))` branch at [rrule.js#L1142](https://github.com/jakubroztocil/rrule/blob/master/lib/rrule.js#L1142)
  * the `else` branch at [rrule.js#L1206](https://github.com/jakubroztocil/rrule/blob/master/lib/rrule.js#L1206)

### CODE REVIEW

**Comment by paulbalomiri** [comment]
Ok, i merged today-s changes into the PR

**Comment by jakubroztocil** [comment]
Thanks @paulbalomiri. 

I'm not completely sure I fully understand the nuances surrounding this issue, but including milliseconds in `Time` seems like the right thing to do.

Would you please make the following changes?

* Use `camelCase` instead of `underscore_separators`.
* Rename *sub second* to  *millisecond*.
* Mention this change in the change log (in README.md)

**Comment by paulbalomiri** [comment]
I summarized the implications on #90 as far as i can tell. But i may be missing sth.

File: README.md [hunk 1]
```diff
@@ -455,10 +455,12 @@ var rule = new RRule(options)
 * * * * *
 
 ### Changelog
-
 * 2.1.0
    * Removed dependency on Underscore.js (thanks @gsf).
    * Various small bugfixes and improvements.
+   * Added Millisecond precision
+     * millisecond offset extracted from dtstart (dtstart.getTime()%1000)
+     * each reccurence is returned with the same offset
```
Comment (jakubroztocil) [inline README.md]: `2.1.0` has already been released. It should go under `2.2.0-dev`.

File: lib/rrule.js [hunk 2]
```diff
@@ -501,9 +504,8 @@ var RRule = function(options, noCache) {
 
     if (!opts.dtstart) {
         opts.dtstart = new Date();
-        opts.dtstart.setMilliseconds(0);
     }
-
+    var millisecondModulo = opts.dtstart.getTime()%1000;
```
Comment (jakubroztocil) [inline lib/rrule.js]: Spaces around ` % ` please

File: lib/rrule.js [hunk 3]
```diff
@@ -1734,31 +1737,31 @@ Iterinfo.prototype.ddayset = function(year, month, day) {
...
     }
     dateutil.sort(set);
     return set;
 };
 
-Iterinfo.prototype.stimeset = function(hour, minute, second) {
-    return [new dateutil.Time(hour, minute, second)];
+Iterinfo.prototype.stimeset = function(hour, minute, second,millisecond) {
+    return [new dateutil.Time(hour, minute, second,millisecond)];
 };
```
Comment (jakubroztocil) [inline lib/rrule.js]: Spaces after `,`

File: tests/tests.js [hunk 4]
```diff
@@ -17,6 +17,8 @@ module("RRule", {
         // NOTE: can take a longer time.
         this.ALSO_TEST_BEFORE_AFTER_BETWEEN = true;
 
+        this.ALSO_TEST_SUBSECOND_PRECISION=true;
+
```
Comment (jakubroztocil) [inline tests/tests.js]: Spaces around ` = ` ;)

**Comment by paulbalomiri** [comment]
Whitespace fun done :)

**Comment by jakubroztocil** [comment]
Don't wanna be a whitespace-nazi, but there're still a few occurrences of missing spaces after a comma :wink:

**Comment by paulbalomiri** [comment]
When do you plan to integrate this, and do you want to release version 2.1.1?
I'm asking because i am actually using the [meteor binding](https://github.com/aramk/rrule/), and would like to know if
* I should make a PR for the binding too
* wait a bit and ask the *aramk* to pull the latest version.

**Comment by paulbalomiri** [comment]
Any chance to have this patch pulled?

**Comment by jkbrzt** [comment]
👍

**Pull request merged**